autogpt/prompts/generator.py

```diff
@@ -135,13 +136,21 @@ def generate_prompt_string(self) -> str:
         Returns:
             str: The generated prompt string.
         """
-        return (
+        base_string = (
             f"Constraints:\n{self._generate_numbered_list(self.constraints)}\n\n"
             "Commands:\n"
             f"{self._generate_numbered_list(self.commands, item_type='command')}\n\n"
             f"Resources:\n{self._generate_numbered_list(self.resources)}\n\n"
             "Performance Evaluation:\n"
             f"{self._generate_numbered_list(self.performance_evaluation)}\n\n"
-            "Respond with only valid JSON conforming to the following schema: \n"
-            f"{llm_response_schema()}\n"
         )
+        config = Config()
+        if config.enable_function_calling:
+            return base_string + (
+                "Respond with only valid JSON conforming to the schema: \n"
```

Comment:
No schema?

Reply:
That's right, the schema is removed from the `messages` of the OpenAI API request, since it is all in the `functions` of the OpenAI API request. Without this fix in `autogpt/prompts/generator.py`, the OpenAI API request response might not contain the proper `args`.